src/app/ReadClient.h

```diff
@@ -323,6 +326,17 @@ class ReadClient : public Messaging::ExchangeDelegate
      */
     CHIP_ERROR SendRequest(ReadPrepareParams & aReadPrepareParams);
 
+    /**
+     *  Re-activate an idle subscription.
+     *
+     *  When subscribing to ICD and liveness timeout reached, the read client will move to the `InactiveICDSubscription` state and
```

Comment:
See, this "will" is I think at the heart of the issue.  The API consumer needs to be able to _choose_ whether it will, or be able to somehow override.

Reply:
updated, thanks